File: Java/LongestIncreasingSubsequence.java
/**
 * The program finds the length of the Longest subsequence (may not be
 * continuous) such that the subsequence is in increasing order
 */
public class LongestIncreasingSubsequence {
	public static void main(String args[]) {
		int arr[] = { 1, 4, 2, 10, 8 };
		int lisLength = findLongestIncSubLength(arr);
		System.out.println("Longest Increasing Subsequence Length is : "+lisLength);
		
	}

	static int findLongestIncSubLength(int arr[]) {
		int length = arr.length;
		int dp[] = new int[length];
		int i, j, max = 0;

		// Initialize Longest Increasing Subsequence values
		for (i = 0; i < length; i++) {
			dp[i] = 1;			
		}
		
		for (i = 1; i < length; i++) {
			for (j = 0; j < i; j++) {
				if (arr[i] > arr[j] && dp[i] < dp[j] + 1) {
					dp[i] = dp[j] + 1;	
				}				
			}
		}
		max = findMax(dp);
		return max;
	}
	
	public static int findMax(int arr[]) {
		int length = arr.length;
		int max = 0;
		for (int i = 0; i < length; i++) {
			if (max < arr[i]) {
				max = arr[i];	
			}			
		}
		return max;
	}
}

File: C/Longest Increasing Subsequence.c
/* Dynamic Programming C implementation of LIS problem */
#include<stdio.h>
#include<stdlib.h>
 
/* lis() returns the length of the longest increasing
  subsequence in arr[] of size n */
int lis( int arr[], int n )
{
    int *lis, i, j, max = 0;
    lis = (int*) malloc ( sizeof( int ) * n );
 
    /* Initialize LIS values for all indexes */
    for (i = 0; i < n; i++ )
        lis[i] = 1;
 
    /* Compute optimized LIS values in bottom up manner */
    for (i = 1; i < n; i++ )
        for (j = 0; j < i; j++ ) 
            if ( arr[i] > arr[j] && lis[i] < lis[j] + 1)
                lis[i] = lis[j] + 1;
 
    /* Pick maximum of all LIS values */
    for (i = 0; i < n; i++ )
        if (max < lis[i])
            max = lis[i];
 
    /* Free memory to avoid memory leak */
    free(lis);
 
    return max;
}
 
/* Driver program to test above function */
int main()
{
    int n,i;
    scanf("%d", &n);
    int arr[n];
    for (i = 0; i < n; i++){
        scanf("%d", &arr[i]);
    }    
    printf("%d \n", lis( arr, n ) );
    return 0;
}
